### ISSUE TITLE
Flameshot Fails to Refresh Multiple Monitor Configuration upon RD Session Switch

### ISSUE THREAD

**Issue by csijake** [flameshot-org/flameshot#2610]
### Flameshot Version

Flameshot v11.0.0 (ad1bf28)

### Installation Type

Using the ready-made package from Github Releases

### Operating System type and version

Windows 11 Business 21H2 (22000.675)

### Description

Flameshot will not recognize multiple monitors when switching between a Remote Desktop session and a direct ("at the computer") session.

Specifically, if Flameshot is installed for the first time within an RD session (one monitor) and then the physical workstation is logged into (multiple monitors), Flameshot will only show the screenshot UI on the main monitor, not the secondary monitors.

**Workaround:** If the *Open Launcher* window is opened, there is code in this part of the app that will detect and "reset" the multiple monitor configuration, so invoking this window will cause screenshot (PrtScr) behavior to work correctly again.

I believe the fix is to listen for a Windows system event for session connect/disconnect and reconfigure/refresh the multiple monitor configuration within Flameshot silently.

### Steps to reproduce

1. On a fresh computer, remote into that computer with Remote Desktop. Ensure RD client is set to "single monitor" only (disable "span multiple monitors" option).
2. Install Flameshot 11 with default options.
3. Invoke the Flameshot UI (PrtScr) and take a screenshot.
4. Close (do NOT log off) of the RD session.
5. Go to the physical computer (with multiple monitors) and log in directly on the workstation.
6. Invoke Flameshot. Notice the UI is limited to the primary monitor only, not the secondary monitors.

Workaround steps from here:

1. Run the "Open Launcher" context menu command.
2. Close the window. (You don't need to save the screenshot.)
3. Press PrtScr again. Notice that now, multiple monitor support is working as intended.

### System Information

Windows 11 Business 21H2 (22000.675)

![image](https://user-images.githubusercontent.com/65626696/168834733-a499d35c-c4f1-4849-8b60-c2cf4c39f387.png)

All monitors are at the same DPI (standard 100% DPI), all monitors are 1920x1200.

**Comment by mmahmoudian** [comment]
> Specifically, if Flameshot is installed for the first time within an RD session (one monitor) and then the physical workstation is logged into (multiple monitors), Flameshot will only show the screenshot UI on the main monitor, not the secondary monitors.

Have you tried killing Flameshot process after installation when you have added more monitors?

**Comment by csijake** [comment]
@mmahmoudian Yes, I believe the program also refreshes its multi-monitor detection upon startup.

The issue is limited to switching from a single-monitor RD session, to a "live" (at the workstation) session with multiple monitors.

**Comment by borgmanJeremy** [comment]
I think this is a duplicate of a couple other bugs. Essentially any time the monitor configuration changes it causes an issue unless flameshot is restarted.

You are on the right track with the fix though, there is a Qt signal that can be monitored for screen events: https://doc.qt.io/qt-5/qguiapplication.html

I just have not prioritized this fix yet.

**Comment by markusdd** [comment]
may I inquire if this has any priority?
I am bugged by this basically daily when switching from my work to my private screen setup and it is very annoying.

I'm afraid I'm not really able to contribute to a fix code wise but it seems the fundamental issue has been recognized already.

So basically just registering interest to get this fixed, because otherwise flameshot is frickin awesome. ❤️

**Comment by andrewljoy** [comment]
This is still an issue on windows, and its very frustrating.

**Comment by borgmanJeremy** [comment]
duplicate of #3597
